### ISSUE TITLE
files.get not working as expected

### ISSUE THREAD

**Issue by mkinney** [Fizzadar/pyinfra#209]
It could be me, but I would expect this:

```
files.get(
    {'Download a file from a remote'},
    '/etc/centos-release',
    '/tmp/whocares',
)
```

to download the file /etc/centos-release and put the contents into a local /tmp/whocares file.

But, I get this error:

```
$ pyinfra -vv 192.168.2.219 --user vagrant --password vagrant files.py
--> Loading config...
--> Loading inventory...

--> Connecting to hosts...
    [192.168.2.219] Connected

--> Preparing operations...
    Loading: files.py
[192.168.2.219] >>> sudo -H sh -c 'ls -ld --time-style=long-iso /etc/centos-release || ls -ldT /etc/centos-release'
[192.168.2.219] -rw-r--r--. 1 root root 38 2019-08-14 06:42 /etc/centos-release
    Loaded fact file: ('/etc/centos-release',)
--> An unexpected exception occurred:

  File "/Users/mikekinney/Documents/python/mkinney_pyinfra/venv/lib/python3.7/site-packages/pyinfra-0.11-py3.7.egg/pyinfra/api/util.py", line 461, in __enter__
    file_io = open(self.filename_or_io, 'rb')
FileNotFoundError: [Errno 2] No such file or directory: '/tmp/whocares'
```

Am I doing something wrong?

**Comment by mkinney** [comment]
More info... (not a local file/perms issue)

```
$ ls -al /tmp/whocares
ls: /tmp/whocares: No such file or directory

$ touch /tmp/whocares

$ rm /tmp/whocares
```

**Comment by Fizzadar** [comment]
This looks like a bug in the download code! Could you paste the contents of `pyinfra-debug.log` here? It should be dumped in the executing directory.

**Comment by mkinney** [comment]
```
$ cat pyinfra-debug.log
  File "/Users/mikekinney/Documents/python/mkinney_pyinfra/venv/lib/python3.7/site-packages/pyinfra-0.11-py3.7.egg/pyinfra_cli/main.py", line 188, in main
    _main(*args, **kwargs)
  File "/Users/mikekinney/Documents/python/mkinney_pyinfra/venv/lib/python3.7/site-packages/pyinfra-0.11-py3.7.egg/pyinfra_cli/main.py", line 464, in _main
    load_deploy_file(state, filename)
  File "/Users/mikekinney/Documents/python/mkinney_pyinfra/venv/lib/python3.7/site-packages/pyinfra-0.11-py3.7.egg/pyinfra_cli/util.py", line 181, in load_deploy_file
    exec_file(filename)
  File "/Users/mikekinney/Documents/python/mkinney_pyinfra/venv/lib/python3.7/site-packages/pyinfra-0.11-py3.7.egg/pyinfra_cli/util.py", line 53, in exec_file
    exec(PYTHON_CODES[filename], data)
  File "files.py", line 58, in <module>
    '/tmp/whocares',
  File "/Users/mikekinney/Documents/python/mkinney_pyinfra/venv/lib/python3.7/site-packages/pyinfra-0.11-py3.7.egg/pyinfra/api/operation.py", line 334, in decorated_func
    **actual_kwargs
  File "/Users/mikekinney/Documents/python/mkinney_pyinfra/venv/lib/python3.7/site-packages/pyinfra-0.11-py3.7.egg/pyinfra/api/util.py", line 212, in unroll_generators
    for item in generator:
  File "/Users/mikekinney/Documents/python/mkinney_pyinfra/venv/lib/python3.7/site-packages/pyinfra-0.11-py3.7.egg/pyinfra/modules/files.py", line 338, in get
    local_sum = get_file_sha1(local_filename)
  File "/Users/mikekinney/Documents/python/mkinney_pyinfra/venv/lib/python3.7/site-packages/pyinfra-0.11-py3.7.egg/pyinfra/api/util.py", line 494, in get_file_sha1
    with file_data as file_io:
  File "/Users/mikekinney/Documents/python/mkinney_pyinfra/venv/lib/python3.7/site-packages/pyinfra-0.11-py3.7.egg/pyinfra/api/util.py", line 461, in __enter__
    file_io = open(self.filename_or_io, 'rb')
FileNotFoundError: [Errno 2] No such file or directory: '/tmp/whocares'
```

**Comment by mkinney** [comment]
Working on adding spellcheck and I think the error is:

`pyinfra/modules/files.py:673:9: SC100 Possibly misspelt word: 'soruce'`

**Comment by mkinney** [comment]
Doh! that was just a comment... nm

**Comment by Fizzadar** [comment]
I've found a couple of bugs in the operation ATM, hoping to finish testing and push today!

**Comment by Fizzadar** [comment]
OK have pushed a bunch of fixes (https://github.com/Fizzadar/pyinfra/commit/61ea96c856e0219e8e3a2dfe5d29b4acd61c9044 + https://github.com/Fizzadar/pyinfra/commit/b4f0d20e6145f9f838ef3ee4284752915bb8fa76 + https://github.com/Fizzadar/pyinfra/commit/8f4108c42d73aeb98e023d2f2b2d422738eeb625 + https://github.com/Fizzadar/pyinfra/commit/fb70efdf3b27f5e64dfea2f2da0b24f268e1aa2c), now tested working with `@local`, SSH and Docker.

Will drop `0.12.1` later this afternoon!

**Comment by Fizzadar** [comment]
Just released `0.12.1` :)
